lib/zephyr/src/zephyr/readers.py

```diff
@@ -113,21 +157,116 @@ def load_parquet(file_path: str, **kwargs) -> Iterator[dict]:
         >>> backend = create_backend("ray", max_parallelism=10)
         >>> ds = (Dataset
         ...     .from_files("/input", "**/*.parquet")
-        ...     .flat_map(load_parquet)
+        ...     .load_parquet()
         ...     .map(lambda r: transform_record(r))
         ...     .write_jsonl("/output/data-{shard:05d}.jsonl.gz")
         ... )
         >>> output_files = list(backend.execute(ds))
     """
-    import pyarrow.parquet as pq
-
-    with open_file(file_path, "rb") as f:
-        # TODO: should we also expose kwargs for ParquetFile constructor?
-        parquet_file = pq.ParquetFile(f)
-        for batch in parquet_file.iter_batches(**kwargs):
+    import pyarrow.dataset as pads
+
+    spec = _as_spec(source)
+    columns = spec.columns
+
+    pa_filter = None
+    if spec.filter_expr is not None:
+        from zephyr.expr import to_pyarrow_expr
+
+        pa_filter = to_pyarrow_expr(spec.filter_expr)
+
```

Comment:
nit: `logger.info("Loading: %s", spec.path)` to match the vortex implementation?